### ISSUE TITLE
make: *** [sub-src-phantomjs-pro-make_first-ordered] Error 2

### ISSUE THREAD

**Issue by ohenak** [ariya/phantomjs#12941]
I am trying to build phantomjs 2.0 to cross-check some issues that I am seeing on 1.9.8.
My Linux version: redhat-release-workstation-6Workstation-6.4.0.4.el6.x86_64

However, `./build.sh` failed at the very beginning of `Building main PhantomJS application...`
```
Building QtWebkit...

cd Source/WTF/ && ( test -e Makefile.WTF || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Source/WTF/WTF.pro WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.WTF ) && make -f Makefile.WTF
make[1]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WTF'
make[1]: Nothing to be done for `first'.
make[1]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WTF'
cd Source/JavaScriptCore/ && ( test -e Makefile.JavaScriptCore || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore/JavaScriptCore.pro WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.JavaScriptCore ) && make -f Makefile.JavaScriptCore
make[1]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore'
( test -e Makefile.LLIntOffsetsExtractor || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore/LLIntOffsetsExtractor.pro WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.LLIntOffsetsExtractor ) && make -f Makefile.LLIntOffsetsExtractor
make[2]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore'
( test -e Makefile.JavaScriptCore.DerivedSources || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore/DerivedSources.pri WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.JavaScriptCore.DerivedSources ) && make -f Makefile.JavaScriptCore.DerivedSources
make[2]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore'
( test -e Makefile.JavaScriptCore.Target || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore/Target.pri WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.JavaScriptCore.Target ) && make -f Makefile.JavaScriptCore.Target
make[2]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore'
( test -e Makefile.jsc || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore/jsc.pro WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.jsc ) && make -f Makefile.jsc
make[2]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore'
make[1]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore'
cd Source/ThirdParty/leveldb/ && ( test -e Makefile.leveldb || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Source/ThirdParty/leveldb/leveldb.pro WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.leveldb ) && make -f Makefile.leveldb
make[1]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/ThirdParty/leveldb'
( test -e Makefile.Target || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Source/ThirdParty/leveldb/Target.pri WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.Target ) && make -f Makefile.Target
make[2]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/ThirdParty/leveldb'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/ThirdParty/leveldb'
make[1]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/ThirdParty/leveldb'
cd Source/WebCore/ && ( test -e Makefile.WebCore || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebCore/WebCore.pro WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.WebCore ) && make -f Makefile.WebCore
make[1]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebCore'
( test -e Makefile.WebCore.DerivedSources || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebCore/DerivedSources.pri WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.WebCore.DerivedSources ) && make -f Makefile.WebCore.DerivedSources
make[2]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebCore'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebCore'
( test -e Makefile.WebCore.Target || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebCore/Target.pri WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.WebCore.Target ) && make -f Makefile.WebCore.Target
make[2]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebCore'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebCore'
make[1]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebCore'
cd Source/WebKit/ && ( test -e Makefile.WebKit1 || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebKit/WebKit1.pro WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.WebKit1 ) && make -f Makefile.WebKit1
make[1]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebKit'
make[1]: Nothing to be done for `first'.
make[1]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebKit'
cd Source/ && ( test -e Makefile.QtWebKit || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Source/QtWebKit.pro WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.QtWebKit ) && make -f Makefile.QtWebKit
make[1]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source'
( test -e Makefile.api || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Source/api.pri WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.api ) && make -f Makefile.api
make[2]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source'
( test -e Makefile.widgetsapi || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Source/widgetsapi.pri WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.widgetsapi ) && make -f Makefile.widgetsapi
make[2]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source'
make[1]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Source'
cd Tools/ && ( test -e Makefile.Tools || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/qt/qtwebkit/Tools/Tools.pro WEBKIT_CONFIG\ -=\ \ use_glib\ use_gstreamer\ use_gstreamer010\ use_native_fullscreen_video\ legacy_web_audio\ web_audio\ video\ gamepad -o Makefile.Tools ) && make -f Makefile.Tools
make[1]: Entering directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Tools'
make[1]: Nothing to be done for `first'.
make[1]: Leaving directory `/home/kho/research/phantomjs/src/qt/qtwebkit/Tools'

Building main PhantomJS application...

cd src/ && ( test -e Makefile.phantomjs || /home/kho/research/phantomjs/src/qt/qtbase/bin/qmake /home/kho/research/phantomjs/src/phantomjs.pro -o Makefile.phantomjs ) && make -f Makefile.phantomjs
make[1]: Entering directory `/home/kho/research/phantomjs/src'
g++ -Wl,-O1 -Wl,-rpath-link,/home/kho/research/phantomjs/src/qt/qtbase/lib -o ../bin/phantomjs phantom.o callback.o webpage.o webserver.o main.o utils.o networkaccessmanager.o cookiejar.o filesystem.o system.o env.o terminal.o encoding.o config.o childprocess.o repl.o crashdump.o mongoose.o linenoise.o utf8.o qcommandline.o minidump_file_writer.o convert_UTF.o md5.o string_conversion.o crash_generation_client.o exception_handler.o log.o linux_dumper.o linux_ptrace_dumper.o minidump_writer.o file_id.o guid_creator.o memory_mapped_file.o safe_readlink.o phantomjs_plugin_import.o qrc_phantomjs.o qrc_ghostdriver.o qrc_WebKit.o qrc_InspectorBackendCommands.o moc_phantom.o moc_callback.o moc_webpage.o moc_webserver.o moc_networkaccessmanager.o moc_cookiejar.o moc_filesystem.o moc_system.o moc_env.o moc_terminal.o moc_config.o moc_childprocess.o moc_repl.o moc_qcommandline.o   -L/home/kho/research/phantomjs/src/qt/qtbase/lib -lQt5WebKitWidgets -L/usr/local/lib/ -Wl,-whole-archive -Wl,-no-whole-archive -L/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebKit/release -Wl,-whole-archive -Wl,-no-whole-archive -L/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebCore/release -Wl,-whole-archive -Wl,-no-whole-archive -L/home/kho/research/phantomjs/src/qt/qtwebkit/Source/ThirdParty/leveldb/release -Wl,-whole-archive -Wl,-no-whole-archive -L/home/kho/research/phantomjs/src/qt/qtwebkit/Source/JavaScriptCore/release -Wl,-whole-archive -Wl,-no-whole-archive -L/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WTF/release -Wl,-whole-archive -Wl,-no-whole-archive -Wl,-whole-archive -Wl,-no-whole-archive -Wl,-whole-archive -Wl,-no-whole-archive -Wl,-whole-archive -Wl,-no-whole-archive -Wl,-whole-archive -Wl,-no-whole-archive -Wl,-whole-archive -Wl,-no-whole-archive -Wl,-whole-archive -Wl,-no-whole-archive -Wl,-whole-archive -Wl,-no-whole-archive -Wl,-whole-archive -Wl,-no-whole-archive -Wl,-whole-archive -Wl,-no-whole-archive -Wl,-whole-archive -Wl,-no-whole-archive -Wl,-whole-archive -Wl,-no-whole-archive -Wl,-whole-archive -Wl,-no-whole-archive -Wl,-whole-archive -Wl,-no-whole-archive -Wl,-whole-archive -Wl,-no-whole-archive -lQt5PrintSupport -lQt5WebKit -Wl,-whole-archive -lWebKit1 -Wl,-no-whole-archive -Wl,-whole-archive -lWebCore -Wl,-no-whole-archive -Wl,-whole-archive -lleveldb -Wl,-no-whole-archive -Wl,-whole-archive -lJavaScriptCore -Wl,-no-whole-archive -Wl,-whole-archive -lWTF -Wl,-no-whole-archive -licudata -lQt5Sql -L/home/kho/research/phantomjs/src/qt/qtbase/plugins/accessible -lqtaccessiblewidgets -lQt5Widgets -L/home/kho/research/phantomjs/src/qt/qtbase/plugins/bearer -lqgenericbearer -lQt5Network -lssl -lcrypto -L/home/kho/research/phantomjs/src/qt/qtbase/plugins/platforms -lqphantom -lQt5PlatformSupport -lfontconfig -lfreetype -L/home/kho/research/phantomjs/src/qt/qtbase/plugins/imageformats -lqico -lQt5Gui -ljpeg -lpng -lqtharfbuzzng -lQt5Core -lz -licui18n -licuuc -lm -ldl -lrt -lpthread
/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebCore/release/libWebCore.a(qrc_WebKit.o): In function `qCleanupResources_WebKit()':
qrc_WebKit.cpp:(.text._Z24qCleanupResources_WebKitv+0x0): multiple definition of `qCleanupResources_WebKit()'
qrc_WebKit.o:qrc_WebKit.cpp:(.text+0x0): first defined here
/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebCore/release/libWebCore.a(qrc_WebKit.o): In function `qInitResources_WebKit()':
qrc_WebKit.cpp:(.text._Z21qInitResources_WebKitv+0x0): multiple definition of `qInitResources_WebKit()'
qrc_WebKit.o:qrc_WebKit.cpp:(.text+0x40): first defined here
/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebCore/release/libWebCore.a(qrc_InspectorBackendCommands.o): In function `qCleanupResources_InspectorBackendCommands()':
qrc_InspectorBackendCommands.cpp:(.text._Z42qCleanupResources_InspectorBackendCommandsv+0x0): multiple definition of `qCleanupResources_InspectorBackendCommands()'
qrc_InspectorBackendCommands.o:qrc_InspectorBackendCommands.cpp:(.text+0x0): first defined here
/home/kho/research/phantomjs/src/qt/qtwebkit/Source/WebCore/release/libWebCore.a(qrc_InspectorBackendCommands.o): In function `qInitResources_InspectorBackendCommands()':
qrc_InspectorBackendCommands.cpp:(.text._Z39qInitResources_InspectorBackendCommandsv+0x0): multiple definition of `qInitResources_InspectorBackendCommands()'
qrc_InspectorBackendCommands.o:qrc_InspectorBackendCommands.cpp:(.text+0x40): first defined here
collect2: ld returned 1 exit status
make[1]: *** [../bin/phantomjs] Error 1
make[1]: Leaving directory `/home/kho/research/phantomjs/src'
make: *** [sub-src-phantomjs-pro-make_first-ordered] Error 2
```

I am following the instructions here:
http://phantomjs.org/build.html

Anyone has experienced this problem?

**Comment by Vitallium** [comment]
This build issue has been fixed in #12899

**Comment by mdproductions** [comment]
Added a compiled binary for 15.10 if anyone interested: https://github.com/mdproductions/phantomjs/tree/master/bin/linux
